### PR TITLE
add Link analytics

### PR DESCRIPTION
**Pull request by benmcginnis**

### CODE REVIEW

File: packages/pages/src/components/analytics/Analytics.ts [hunk 1]
```diff
@@ -179,7 +180,7 @@ export class Analytics implements AnalyticsMethods {
   trackClick(
     eventName: string,
     conversionData?: ConversionDetails
-  ): (e: MouseEvent) => Promise<void> {
+  ): (e: MouseEvent<HTMLAnchorElement>) => Promise<void> {
```
Comment (bhainesva) [inline packages/pages/src/components/analytics/Analytics.ts:183]: Should we also support button elements?

File: packages/pages/src/components/link/link.tsx [hunk 2]
```diff
@@ -86,17 +91,28 @@ export const Link = React.forwardRef<HTMLAnchorElement, LinkProps>(
     const { children, onClick, className, ...rest } = props;
     const link: CTA = isHREFProps(props) ? { link: props.href } : props.cta;
 
+    let trackEvent: string|null;
+    let analytics: AnalyticsMethods|null;
+
+    if (AnalyticsContext !== null) {
+      trackEvent = props.eventName || props.cta ? "cta" : "link";
+      analytics = useAnalytics();
```
Comment (bhainesva) [inline packages/pages/src/components/link/link.tsx:99]: hooks can only be called at the top level of the component: https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

File: packages/pages/src/components/analytics/Analytics.ts (hunk 1 above)
Comment (benmcginnis) [reply packages/pages/src/components/analytics/Analytics.ts:183]: so i was wondering about this, `buttons` should not navigate and trackClick won't actually work if you put it on a button as it needs the `href` to figure out what it's going to do. As it stands right now if there is now `href` it behaves just like `track()` so people should really only use this on an `a` element.

Comment (benmcginnis) [reply packages/pages/src/components/analytics/Analytics.ts:183]: so i guess i'm saying, No, a button should use `track` and not `trackClick`

File: packages/pages/src/components/link/link.tsx (hunk 2 above)
Comment (benmcginnis) [reply packages/pages/src/components/link/link.tsx]: probably should add the hooks es lint rule

**Pull request merged**